### ISSUE TITLE
`dbWriteTable()` requires `row.names` argument

### ISSUE THREAD

**Issue by ibecav** [r-dbi/odbc#737]
There seems to be a change in the behavior of `dbWriteTable()` that is unexpected and seems very clunky, regarding `row.names`.

If the user wants to use `dbWriteTable()` to append to an existing table they must now explicitly set the `row.names` in addition to `append` and `overwrite`.  Seems to be undocumented if that's the desired behavior and the error it throws is pretty confusing.

**N.B.** reprex is using odbc_1.4.1.9000 and DBI_1.2.1 but seems unrelated to the odbc version

``` r
library(reprex)
library(DBI)

con <- DBI::dbConnect(odbc::odbc(), "best")
name <- DBI::Id(
  database = "DPH_COVID_IMPORT",
  schema = "dbo",
  table = "junkpleasedelete2"
)

dbCreateTable(con, name, mtcars)
dbExistsTable(con, name)
#> [1] TRUE
dbAppendTable(con, name, mtcars)
dbWriteTable(con, name, mtcars) # Expected behavior
#> Error: Table <Id> "DPH_COVID_IMPORT"."dbo"."junkpleasedelete2" exists in database, and both overwrite and append are FALSE
dbWriteTable(con, name, mtcars, 
             append = TRUE, overwrite = FALSE) # Unexpected behavior
#> Error in eval(expr, envir, enclos): nanodbc/nanodbc.cpp:1771: 42000
#> [Microsoft][SQL Server Native Client 11.0][SQL Server]Invalid column name 'row_names'. 
#> [Microsoft][SQL Server Native Client 11.0][SQL Server]Statement(s) could not be prepared.
dbWriteTable(con, name, mtcars, 
             append = TRUE, overwrite = FALSE, 
             row.names = FALSE) # Feels very unnecessary but works 

# cleanup
dbRemoveTable(con, name)
```

<details>

```R
sessionInfo()
#> R version 4.3.2 (2023-10-31 ucrt)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows Server 2016 x64 (build 14393)
#> 
#> Matrix products: default
#> 
#> 
#> locale:
#> [1] LC_COLLATE=English_United States.1252 
#> [2] LC_CTYPE=English_United States.1252   
#> [3] LC_MONETARY=English_United States.1252
#> [4] LC_NUMERIC=C                          
#> [5] LC_TIME=English_United States.1252    
#> 
#> time zone: America/New_York
#> tzcode source: internal
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] DBI_1.2.1    reprex_2.1.0
#> 
#> loaded via a namespace (and not attached):
#>  [1] digest_0.6.34     fastmap_1.1.1     bit_4.0.5         xfun_0.41        
#>  [5] odbc_1.4.1.9000   glue_1.6.2        blob_1.2.4        knitr_1.45       
#>  [9] pkgconfig_2.0.3   htmltools_0.5.7   bit64_4.0.5       rmarkdown_2.25   
#> [13] lifecycle_1.0.4   cli_3.6.2         vctrs_0.6.5       withr_3.0.0      
#> [17] compiler_4.3.2    rstudioapi_0.15.0 tools_4.3.2       hms_1.1.3        
#> [21] evaluate_0.23     Rcpp_1.0.11       yaml_2.3.8        rlang_1.1.2      
#> [25] fs_1.6.3
```
</details>

<sup>Created on 2024-01-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by krlmlr** [comment]
Thanks. Transferred because odbc implements its own `dbWriteTable()` .

**Comment by hadley** [comment]
@ibecav FYI you generally don't need to supply `sessionInfo()` unless we specifically ask for it, and in most cases the only thing you need to show in the reprex is the problem.

**Comment by hadley** [comment]
Fixed by #691
